Cannot compute unary functions of simple ADScalar type

Calling unary functions like sin, cos, log on a simple ADScalar value doesn't currently work, e.g:

```cpp
#define T double
ADScalar<T, 1> xScalar(1), y;
y = sin(xScalar);
```

This fails with:

```
/home/ali/repos/a2d/include/adscalar.h:326:22: error: no matching function for call to 'sin'
  326 |   ADScalar<X, M> out(sin(r.value));
      |                      ^~~
a2d-learning.cpp:39:7: note: in instantiation of function template specialization 'A2D::sin<double, 1>' requested here
   39 |   y = sin(xScalar);
      |       ^
/home/ali/repos/a2d/include/ad/a2dunary.h:192:24: note: candidate template ignored: could not match 'const ADExpr<A, Ta>' against 'const double'
  192 | A2D_1ST_UNARY(SinExpr, sin, std::sin(a.value()), std::cos(a.value()), tmp)
      |                        ^
/home/ali/repos/a2d/include/ad/a2dunary.h:192:24: note: candidate template ignored: could not match 'ADExpr<A, Ta>' against 'const double'
/home/ali/repos/a2d/include/ad/a2dunary.h:273:25: note: candidate template ignored: could not match 'const A2DExpr<A, Ta>' against 'const double'
  273 | A2D_2ND_UNARY(SinExpr2, sin, std::sin(a.value()), std::cos(a.value()), tmp,
      |                         ^
/home/ali/repos/a2d/include/ad/a2dunary.h:273:25: note: candidate template ignored: could not match 'A2DExpr<A, Ta>' against 'const double'
/home/ali/repos/a2d/include/adscalar.h:325:23: note: candidate template ignored: could not match 'const ADScalar<X, M>' against 'const double'
  325 | inline ADScalar<X, M> sin(const ADScalar<X, M> &r) {
      |
 ```

--- a/include/a2ddefs.h
+++ b/include/a2ddefs.h
@@ -265,6 +265,16 @@ A2D_FUNCTION T log(T val) {
   return std::log(val);
 }
 
+template <typename T, std::enable_if_t<is_scalar_type<T>::value, bool> = true>
+A2D_FUNCTION T sin(T val) {
+  return std::sin(val);
+}
+
+template <typename T, std::enable_if_t<is_scalar_type<T>::value, bool> = true>
+A2D_FUNCTION T cos(T val) {
+  return std::cos(val);
+}
+
 template <class ForwardIt, class T>
 void fill(ForwardIt first, ForwardIt last, const T& value) {
   std::fill(first, last, value);

--- a/include/adscalar.h
+++ b/include/adscalar.h
@@ -328,6 +328,7 @@ inline ADScalar<X, M> sin(const ADScalar<X, M> &r) {
   for (int i = 0; i < M; i++) {
     out.deriv[i] = d * r.deriv[i];
   }
+  return out;
 }
 
 template <class X, int M>
@@ -337,6 +338,7 @@ inline ADScalar<X, M> cos(const ADScalar<X, M> &r) {
   for (int i = 0; i < M; i++) {
     out.deriv[i] = d * r.deriv[i];
   }
+  return out;
 }
 
 }  // namespace A2D

